scripts/check_docs_formatted.py

```diff
@@ -262,16 +292,22 @@ def main(argv: Sequence[str] | None = None) -> int:
             print("Please remove them and re-run.")
             return 1
 
+    # Build ruff
+    print("Building ruff, this may take a some time...")
+    subprocess.run(["cargo", "build"], cwd=ROOT_DIR, check=True)
+    print("Ruff build complete.")
+
```

Comment:
Should we instead use the latest released version? That way we don't need to wait for the build to complete. This would be added to the requirements file.

Reply:
My reason for not using the latest released version of Ruff was to avoid the situation where an update to the formatter results in an update to the docs. This could potentially result in the docs being formatted differently from the current release version of ruff and would only be fixed in a subsequent PR and release; however, using the current branch's version of the formatter allows the docs to be updated at the same time as the formatter by someone who has the context of the change. 

If this is not a concern, let me know and I can update to use the latest release.

Reply:
I think any major style updates would only go out in a breaking release and will be stabilized once per year (similar to how Black does it). So, it shouldn't be too much work to update the docs then ;)

Reply:
Updated in: b3ac44d92ca62e3f6cc7096c565eeaca994aab1b